bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/Rule.java

```diff
@@ -101,6 +104,30 @@ public String getUID() {
     }
 
     /**
+     * Gets the scope of the Rule, if any exist. This property is either set by the {@link RuleEngine} when
+     * the {@link Rule} is added or by the creating party. It's an optional property.
+     * <br/><br/>Implementation note
+     *    <br/>The scope is part of the {@code UID} and the prefix thereof.
+     *    <br/>If the UID does not contain a {@link SCOPE_DELIMITER} {@code null} will be returned.
+     *    <br/>If the UID does contain a {@link SCOPE_DELIMITER} the prefix until the first occurrence will be returned.
+     *    <br/>If the prefix would have a zero length {@code null} will be returned.
+     *
+     * @return scope of this {@link Rule}, or {@code null} if no scope or an empty scope is found
+    */
+    public String getScope() {
```

Comment:
My understanding was that the scope wouldn't be a field of its own, so that persistence and other pats of the code don't have to be changed.
However it would be very useful to have the `getScope()` within the `Rule` class so that it can be used as part of the API.
For same reason it would be great to include the scope within the ID creation of the `RuleEngine` so that `Rule`s created by the RuleEngine can be identified as such.
The idea is to have `getScope()` in the API because otherwise every client would have to implement it itself.
In contrast to tags this property ("scope") cannot be edited by the user and therefor is a strong indicator "who" created the rule and can be used by specialzed clients of the API.